_posts/2018-05-18-contributing-to-installer-puppet-modules.md

```diff
@@ -0,0 +1,326 @@
+---
+layout: post
+title: 'Contributing to the Installer Puppet Modules'
+date: 2018-05-21 12:00:00
+author: Christopher Duryee
+tags:
+- foreman
+---
+---
+layout: post
+title: 'Contributing to the Installer Puppet Modules'
+date: 2018-05-21 12:00:00
+author: Christopher Duryee
+tags:
+- foreman
+---
+
+# How do I get started with contributing to the installer?
+
+The Foreman installer can be intimidating at first. It manages lots of files, and has many, many, many options. Even more options are exposed via `custom-hiera.yaml`. If you want to make a change that's not covered by either an installer option or `custom-hiera.yaml`, you may feel like you are out of luck. However, this is a great oppurtunity to make an upstream contribution!
+
+This blog post shows the steps on how to identify a change you want to make in the installer and create an upstream pull request in the correct installer module.
+
```

Comment:
Is it worth clarifying that we regard the installer as the "official" distribution, and we are probably not going to merge PRs for edgy plugins that aren't well tested? In fact, should we say that in the main docs too (I feel it's how we *actually* work, but I'm not sure it's documented anywhere...)

Reply:
Do you mean like PRs that plugins to the installer?

Reply:
Apologies, missed this one @beav. I do mean that, yes - I feel we need to do *some* curation at some level, but since it makes sense to have plugins under theforeman namespace in general, the installer is the right place to do the curation?